sklearn/utils/validation.py

```diff
@@ -506,14 +507,38 @@ def check_array(array, accept_sparse=False, dtype="numeric", order=None,
         msg = ("Data with input dtype %s was converted to %s%s."
                % (dtype_orig, array.dtype, context))
         warnings.warn(msg, DataConversionWarning)
+
+    # Indices Datatype regulation
+    _check_large_sparse(array, accept_large_sparse)
+
     return array
 
 
+def _check_large_sparse(array, accept_large_sparse=False):
+    """
+    Indices Regulation for CSR Matrices.
+    Only Int32 are supported for now
+
+    """
+    if accept_large_sparse is False and type(array) == sp.csr.csr_matrix:
+        supported_indices = ["int32"]
+        if (hasattr(array, "indices") and hasattr(array.indices, "dtype") and
+                array.indices.dtype not in supported_indices):
+            raise ValueError("CSR arrays accepts only 32 bit integer indexing."
+                             " You provided % s bit indexing"
+                             % array.indices.dtype)
+        if (hasattr(array, "indptr") and hasattr(array.indptr,  "dtype") and
+                array.indptr.dtype not in supported_indices):
+            raise ValueError("CSR arrays accepts only 32 bit integer indexing."
+                             " You provided %s bit indexing"
+                             % array.indices.dtype)
+
+
 def check_X_y(X, y, accept_sparse=False, dtype="numeric", order=None,
               copy=False, force_all_finite=True, ensure_2d=True,
               allow_nd=False, multi_output=False, ensure_min_samples=1,
               ensure_min_features=1, y_numeric=False,
-              warn_on_dtype=False, estimator=None):
+              warn_on_dtype=False, estimator=None, accept_large_sparse=False):
```

Comment:
This needs documentation under Parameters. I also wonder whether we should just be some alteration to `accept_sparse`, e.g. having `accept_sparse=['csr', 'csr64', ...]` means that a 32bit-indexed or a 64bit-indexed csr is accepted.

Reply:
I feel this should be different variable altogether

cuz in later phases, when 64bits indices will be supported  (less iteration would be required to fix)